### DISCUSSION TITLE
My Vaultwarden is always out of sync

### DISCUSSION THREAD

**Discussion by Anas-1554** [dani-garcia/vaultwarden#4181] [Q&A]
I checked my server timezone it is same to my time, Code had about 15 sec left before exipre can you tell what I can do to fix this issue?
![image](https://github.com/dani-garcia/vaultwarden/assets/94738878/664830ce-bcd7-49ff-8fdb-2a5a00861d55)

**Comment by Anas-1554** [comment]
Hello

**Comment by BlackDex** [comment]
Did you checked `/admin/diagnostics`?

**Comment by Anas-1554** [comment]
Everything is good their

**Comment by BlackDex** [comment]
And the time, that is exact the same for all devices and the server if you check that page?

**Comment by Anas-1554** [comment]
Yes, My PC and all of the device have same time as server.

**Comment by BlackDex** [comment]
Then i have no clue. Those tokens are 30seconds valid per 0 and 30 second. If you didn't disable that a previous and next token are invalid, then it should just work.

**Comment by Anas-1554** [comment]
![image](https://github.com/dani-garcia/vaultwarden/assets/94738878/8fcdd1ea-c318-451e-b9ca-aec6cb45dce3)
This is the settings

**Reply by BlackDex** [reply]
That has nothing to do with a TOTP authenicator token

**Comment by stefan0xC** [comment]
How do you generate the TOTP? And given that the error message is not from Vaultwarden (and also not the Bitwarden Web Vault) I'm assuming that you are generating the code either via the web-vault, browser extension or the mobile app and it looks like the third party server (where you are trying to authenticate) is rejecting the TOTP. Are you sure that that server has set the correct time? 

Also the NTP check from the `/admin/diagnostics` page will return Ok if it's within 15 seconds so it might not be the most reliable (if the code sometimes works and not always). Can you check how much the device`s clock is off (specifically the device you are using to generate the TOTP) with a site like [time.gov](https://time.gov) or [uhr.ptb.de](https://uhr.ptb.de)?

**Reply by BlackDex** [reply]
And, don't forget that if it's generated on a mobile device and that device is out-of-sync with ntp, it will also cause issues.

**Reply by Anas-1554** [reply]
On the website I get TOTP code then put it into Vaultwarden, I check time of server and my PC still getting 2FA issue, Can't see any error in `/admin/diagnostics`

**Comment by Anas-1554** [comment]
![image](https://github.com/dani-garcia/vaultwarden/assets/94738878/1bf71d9e-5f8e-454e-aaa0-574a697bd74f)

Same issue of diffrent website even after sync

**Reply by BlackDex** [reply]
please describe exactly step-by-step what you are doing and which application and how.
It looks to me like the device you are using which provides the TOTP tokens you enter that device is probably out-of-sync.

To clarify Vaultwarden it self doesn't generate those tokens, its always the clients which do that. And they do that based upon the time of the client device.

So, if the TOTP tokens generated are not correct, then the date/time on your device is not correct.

**Comment by adarme47** [comment]
Hi,
I think i have the same problem.
for couple days, i can't synchronise anymore the extension from my web browsers (vivaldi or chrome) "echec de synchronisation", and i can t anymore re-connect the extension (after re-installation) to my account "une erreur est survenue cannot read properties of null (reading'iterations).

all is ok with my android app.

i do not made any changement of configuration of anything. 

best regards
david

**Reply by BlackDex** [reply]
I think it is a totally different issue you have. Be sure you updated to the latest version of Vaultwarden, and also fully logout and back in again, not lock, but logout.

**Reply by adarme47** [reply]
thank u for your answer.
could u tell me were i can find the way to update on synology container manager the vaultwarden/server:latest packet to avoid to lose my configuration?
best regards david

**Reply by BlackDex** [reply]
I have no clue how those devices work so can't help you there.
Just be sure to check the version you are running right now via `/admin/diagnostics`

**Comment by Anas-1554** [comment]
#1: Let's try to re sync the code

- Visit Payapl.com.
- Got to profile settings.
- Enable 2FA code. In my case. I already have one then I will click "update".
- Now you will see QR with code and below that you will see random string copy it and put it in vaultwarden it will give you 2FA code
- Use it on the website and done

**Comment by Anas-1554** [comment]
This work fine for my issue happes after 6/7 days when I have to resync this code.

**Comment by BlackDex** [comment]
I'm for one unsure why you have to resync your papal code, but ok.
Second, i still do not see what the issue is with Vaultwarden, you didn't explained at all what Vaultwarden is reporting or what you do with Vaultwarden when it breaks.

All I can say here is. Fully logout of all your clients, not lock, login again and try again.
